### PR TITLE
fix: Stabilize slideover modal

### PR DESCRIPTION
**Pull request by andrewdwallo**
This is an update (part 2) to stabilize the slideover modal to prevent jittery behavior on open. The original commit was reverted. This commit is to find common ground on what would be best as previously discussed. The original commit and all relevant information: #10713

### CODE REVIEW

**Comment by binaryfire** [comment]
Thanks for working on this! I've been noticing this issue as well.

File: packages/support/resources/views/components/modal/index.blade.php [hunk 1]
```diff
@@ -147,7 +146,7 @@
                 x-transition:leave="duration-300"
                 @if ($width === MaxWidth::Screen)
                 @elseif ($slideOver)
-                    x-transition:enter-start="translate-x-full rtl:-translate-x-full"
+                    x-transition:enter-start="translate-x-2/3 rtl:-translate-x-2/3"
```
Comment (zepfietje) [inline packages/support/resources/views/components/modal/index.blade.php:149]: Could you elaborate on how this fixes the issue, @andrewdwallo?
Also, does adding `will-change` here make any difference?

Comment (andrewdwallo) [reply packages/support/resources/views/components/modal/index.blade.php:149]: Hi @zepfietje,

The change from `translate-x-full` to `translate-x-2/3` reduces the off-screen movement of the modal. This adjustment is particularly beneficial in browsers not utilizing hardware acceleration, as CPU rendering can exacerbate jitteriness. Even with hardware acceleration, this modification contributes to a smoother transition, as the issue was still present.

Regarding the removal of `will-change: transform`, it was eliminated for several reasons:
1. **No Direct Transformation:** It was applied to the outer div, which didn't undergo direct transformations. `will-change` is intended to optimize elements that experience transformations. Its use where no transformations occur can mislead the browser's optimization strategies.
2. **Non-Application to Child Elements:** This property does not inherently apply to an element's children. The actual transformations were on the modal's children, not the outer div, rendering the use of `will-change: transform` on the outer div ineffective.
3. **Performance Considerations:** `will-change` should be used sparingly, as a last resort, because it hints to the browser to prepare for changes. Overuse or misuse can lead to resource mismanagement. Removing it from elements where it's not beneficial ensures more efficient resource use.
4. **Inappropriate Static Use:** The property was originally statically applied. Best practices recommend applying `will-change` dynamically before an element undergoes a transformation or transition, and removing it afterward. This approach prevents the browser from holding onto unnecessary optimizations and aligns with efficient resource management.
5. **Other Benefits:** Eliminating unnecessary use of `will-change` also helps in preventing side effects like creating new layers that could complicate the rendering process, thus maintaining a leaner and more efficient rendering pipeline.

In summary, these adjustments, both in the transition properties and the removal of `will-change: transform`, align with best practices for performant animations and transitions. They ensure the browser's optimizations are effectively and appropriately used.

I hope this clarifies the modifications made in the pull request.

Comment (andrewdwallo) [reply packages/support/resources/views/components/modal/index.blade.php:149]: Also, even if `will-change: transform` was applied to the correct element, there would most likely be no benefit. I tested it myself and saw no improvement, so it can depend, but it is best to not use it at all. Using `will-change: transform` on elements with transformations is often unnecessary as modern browsers already apply optimal rendering practices. Overriding this with `will-change` can be costly, potentially leading to unnecessary resource allocation.

**Comment by danharrin** [comment]
I think the `will-change: transform` needs to stay, for #6102

**Comment by andrewdwallo** [comment]
@danharrin Adding `will-change: transform` here did not fix that issue. Actually, that issue is still present currently. I just tested it. `will-change: transform` here is not affecting that in any manner. I can try to find out the cause of #6102 and #10926 and try to fix it though if you would like?

**Comment by andrewdwallo** [comment]
I found the fix to #6102 and #10926. I will make a separate PR here shortly.

**Comment by danharrin** [comment]
Thanks!

**Review by zepfietje — CHANGES_REQUESTED** [review]
Let me know once this is ready for final review, @danharrin @andrewdwallo.

Probably want to add an opacity translation as well, since otherwise the slide-over modal would just appear and hide with the horizontal animation. Or is it already present?

**Comment by andrewdwallo** [comment]
I'm not 100% sure, but I think specifically for the slideover, adding an opacity transition (fading in and out), will make it lose that sliding effect and instead will appear as if it’s simply fading into view more..

**Comment by andrewdwallo** [comment]
I will check tomorrow though after I get some sleep. 👍

**Comment by andrewdwallo** [comment]
I think it is good. I also did want to mention that removing the opacity from `x-transition.duration.300ms.opacity` has a pretty major effect on what is actually occurring. I think somehow the transform scale and origin is appearing in the center of the page and then quickly positions in its place to produce the slideover effect. Not sure if this is intended behavior though...

**Comment by andrewdwallo** [comment]
Hey guys, I was hoping for some feedback about this PR. Is there a reason why this is being delayed from being merged? I don't really see any negative consequences from the changes made in this PR, and I actually see only positive aspects of merging this considering that many users have also pointed out this issue.. Is there something else that needs to be worked on first before this? Is it an underlying alpine issue or something else?

**Comment by zepfietje** [comment]
I'm looking into this now, @andrewdwallo.
The reason for it not being merged yet is simply that I have limited time to work on Filament (I'd love to be able to spend more time on it...).

Will keep you updated with my findings.

**Comment by zepfietje** [comment]
I've fixed the modal animation issue in https://github.com/filamentphp/filament/commit/66f9f7d772e1197462aebadd166c8e38038e40ba.
It was caused by overflow being hidden on a wrapping element.
